homeassistant/components/sensor/phicomm.py

```diff
@@ -0,0 +1,228 @@
+"""
+Support for Phicomm air sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.phicomm/
+"""
+
+import asyncio
+import logging
+
+from datetime import timedelta
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, CONF_USERNAME, CONF_PASSWORD,
+    CONF_SENSORS, CONF_SCAN_INTERVAL, TEMP_CELSIUS)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+AUTH_CODE = 'feixun.SH_1'
+TOKEN_FILE = '.phicomm.token.'
+USER_AGENT = 'zhilian/5.7.0 (iPhone; iOS 10.0.2; Scale/3.00)'
+
+TOKEN_URL = 'https://accountsym.phicomm.com/v1/login'
+DATA_URL = 'https://aircleaner.phicomm.com/aircleaner/getIndexData'
+
+SENSOR_PM25 = 'pm25'
+SENSOR_HCHO = 'hcho'
+SENSOR_TEMPERATURE = 'temperature'
+SENSOR_HUMIDITY = 'humidity'
+
+DEFAULT_NAME = 'Phicomm'
+DEFAULT_SENSORS = [SENSOR_PM25, SENSOR_HCHO,
+                   SENSOR_TEMPERATURE, SENSOR_HUMIDITY]
+
+SENSOR_MAP = {
+    SENSOR_PM25: ('PM2.5', 'μg/m³', 'blur'),
+    SENSOR_HCHO: ('HCHO', 'mg/m³', 'biohazard'),
+    SENSOR_TEMPERATURE: ('Temperature', TEMP_CELSIUS, 'thermometer'),
+    SENSOR_HUMIDITY: ('Humidity', '%', 'water-percent')
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SENSORS, default=DEFAULT_SENSORS):
+        vol.All(cv.ensure_list, vol.Length(min=1), [vol.In(SENSOR_MAP)]),
+    vol.Optional(CONF_SCAN_INTERVAL, default=timedelta(seconds=120)): (
+        vol.All(cv.time_period, cv.positive_timedelta)),
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
```

Comment:
HASS requires at least python 3.5.3, so it is safe to use the async/await syntax